src/htm/regions/DatabaseOutRegion.cpp

```diff
@@ -0,0 +1,355 @@
+
+DatabaseOutRegion::DatabaseOutRegion(const ValueMap &params, Region* region)
+    : RegionImpl(region), filename_(""),
+	  dbHandle(nullptr),xTransactionActive(false) {
+  if (params.contains("outputFile")) {
+    std::string s = params.getString("outputFile", "");
+    openFile(s);
+  }
+  else
+    filename_ = "";
+
+}
+
+DatabaseOutRegion::DatabaseOutRegion(ArWrapper& wrapper, Region* region)
+    : RegionImpl(region), filename_(""),
+	  dbHandle(nullptr),xTransactionActive(false) {
+  cereal_adapter_load(wrapper);
+}
+
+
+DatabaseOutRegion::~DatabaseOutRegion() { closeFile(); }
+
+void DatabaseOutRegion::initialize() {
+  NTA_CHECK(region_ != nullptr);
+  // We have no outputs or parameters; just need our input.
+  const std::map<std::string, std::shared_ptr<Input>> inputs = region_->getInputs();
+
+
+  NTA_ASSERT(inputs.size()!=0) << "DatabaseOutRegion::initialize - no inputs configured\n";
+
+  iTableCount = 0;
+	for (const auto & inp : inputs) {
+		const auto inObj = inp.second;
+		if (inObj->hasIncomingLinks() && inObj->getData().getCount() != 0) { //create tables only for those, whose was configured
+			createTable("dataStream_" + inp.first);
+			iTableCount++;
+		}
+	}
+
+}
+
+void DatabaseOutRegion::createTable(const std::string &sTableName){
+
+	/* Create SQL statement */
+	std::string  sql = "CREATE TABLE "+sTableName+" (iteration INTEGER PRIMARY KEY, value REAL);";
+
+	char *zErrMsg;
+	/* Execute SQL statement */
+	int returnCode = sqlite3_exec(dbHandle, sql.c_str(), nullptr, 0, &zErrMsg);
+
+	if( returnCode != SQLITE_OK ){
+		NTA_THROW << "Error creating SQL table, message:"
+				  << zErrMsg;
+		sqlite3_free(zErrMsg);
+	}
+}
+
+void DatabaseOutRegion::insertData(const std::string &sTableName, const std::shared_ptr<Input> inputData){
+
+	NTA_ASSERT(inputData->getData().getCount()==1);
+
+	if(!xTransactionActive){
+		//starts transaction, for speedup. Transaction ends by closing the file
+		ExecuteSQLcommand("BEGIN TRANSACTION");
+		xTransactionActive = true;
+	}
+
+	Real32* value = (Real32 *)inputData->getData().getBuffer();
+	/* Create SQL statement */
+	std::string  sql = "INSERT INTO "+sTableName+"(value) VALUES ("+std::to_string(*value) + ");";
+
+	char *zErrMsg;
+
+	int returnCode = sqlite3_exec(dbHandle, sql.c_str(), nullptr, 0, &zErrMsg);
+
+	if( returnCode != SQLITE_OK ){
+		NTA_THROW << "Error inserting data to SQL table, message:"
+				  << zErrMsg;
+		sqlite3_free(zErrMsg);
+	}
+}
+
+void DatabaseOutRegion::compute() {
+
+
+	const std::map<std::string, std::shared_ptr<Input>> inputs = region_->getInputs();
+
+
+	NTA_ASSERT(inputs.size()!=0) << "DatabaseOutRegion::initialize - no inputs configured\n";
+
+	for (const auto & inp : inputs)
+	{
+		const auto inObj = inp.second;
+		if (inObj->hasIncomingLinks() && inObj->getData().getCount() != 0) { //create tables only for those, whose was configured
+			insertData("dataStream_" + inp.first, inObj);
+		}
+	}
+}
```

Comment:
Yes i think it can be used for reading from the database with not much effort, but i see no use-case now. That's also why it would be good to name the region "DatabaseRegion" as you suggest.